File: backend/app/schemas/ticket.py
from datetime import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class ETicketRead(BaseModel):
    ticket_id: int
    ticket_config_id: int
    event_day_id: int
    booking_id: Optional[int] = None
    ticket_code: Optional[str] = None
    ticket_status: str
    row_label: Optional[str] = None
    col_number: Optional[int] = None
    ticket_type: Optional[str] = None
    price: Optional[Decimal] = None
    event_name: Optional[str] = None
    venue_name: Optional[str] = None
    date: Optional[datetime] = None
    customer_name: Optional[str] = None
    qr_code_url: Optional[str] = None
    issued_at: Optional[datetime] = None
    used_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}

File: frontend/src/types/ticket.ts
export type TicketStatus = "Available" | "Holding" | "Valid" | "Used" | "Canceled" | "Refunded" | "Reserved" | string;

export type ETicket = {
  ticket_id: number;
  ticket_config_id: number;
  event_day_id: number;
  booking_id?: number | null;
  ticket_code?: string | null;
  ticket_status: TicketStatus;
  row_label?: string | null;
  col_number?: number | null;
  ticket_type?: string | null;
  price?: string | number | null;
  event_name?: string | null;
  venue_name?: string | null;
  date?: string | null;
  customer_name?: string | null;
  qr_code_url?: string | null;
  issued_at?: string | null;
  used_at?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
};
